Code/GraphMol/FindRings.cpp

```diff
@@ -745,15 +747,17 @@ bool _atomSearchBFS(const ROMol &tMol, unsigned int startAtomIdx,
     while (nbrIdx != endNbrs) {
       if (*nbrIdx == endAtomIdx) {
         if (currAtomIdx != startAtomIdx) {
-          tv.push_back(rdcast<unsigned int>(*nbrIdx));
+          INT_VECT nv(tv);
+
+          nv.push_back(rdcast<unsigned int>(*nbrIdx));
           // make sure the ring we just found isn't already in our set
           // of rings (this was an extension of sf.net issue 249)
           boost::uint32_t invr =
-              RingUtils::computeRingInvariant(tv, tMol.getNumAtoms());
+              RingUtils::computeRingInvariant(nv, tMol.getNumAtoms());
           if (invars.find(invr) == invars.end()) {
             // we're done!
-            res.resize(tv.size());
-            std::copy(tv.begin(), tv.end(), res.begin());
+            res.resize(nv.size());
```

Comment:
Nope, because then `res` is modified even if the function returns false.
Besides it's saving one copy (the one at line 760) out of many in a function that's not called very much.